java/src/main/java/ai/onnxruntime/OrtSession.java

```diff
@@ -0,0 +1,494 @@
+            setOptimizationLevel(OnnxRuntime.ortApiHandle,nativeHandle, level.getID());
+        }
+
+        /**
+         * Sets the size of the CPU thread pool used for executing multiple request concurrently, if executing on a CPU.
+         * @param numThreads The number of threads to use.
+         * @throws OrtException If there was an error in native code.
+         */
+        public void setInterOpNumThreads(int numThreads) throws OrtException {
+            setInterOpNumThreads(OnnxRuntime.ortApiHandle,nativeHandle,numThreads);
+        }
+
+        /**
+         * Sets the size of the CPU thread pool used for executing a single graph, if executing on a CPU.
+         * @param numThreads The number of threads to use.
+         * @throws OrtException If there was an error in native code.
+         */
+        public void setIntraOpNumThreads(int numThreads) throws OrtException {
+            setIntraOpNumThreads(OnnxRuntime.ortApiHandle,nativeHandle,numThreads);
+        }
+
+        /**
+         * Add CUDA as an execution backend, using device 0.
+         */
+        public void addCUDA() throws OrtException {
+            addCUDA(0);
+        }
+
+        /**
+         * Add CUDA as an execution backend, using the specified CUDA device id.
+         * @param deviceNum The CUDA device id.
+         */
+        public void addCUDA(int deviceNum) throws OrtException {
+            addCUDA(OnnxRuntime.ortApiHandle,nativeHandle,deviceNum);
+        }
+
+        public void addCPU(boolean useArena) throws OrtException {
+            addCPU(OnnxRuntime.ortApiHandle,nativeHandle,useArena?1:0);
+        }
+
+        public void addMkldnn(boolean useArena) throws OrtException {
+            addMkldnn(OnnxRuntime.ortApiHandle,nativeHandle,useArena?1:0);
+        }
+
+        public void addNGraph(String ngBackendType) throws OrtException {
+            addNGraph(OnnxRuntime.ortApiHandle,nativeHandle,ngBackendType);
+        }
+
+        public void addOpenVINO(String deviceId) throws OrtException {
+            addOpenVINO(OnnxRuntime.ortApiHandle,nativeHandle,deviceId);
+        }
+
+        public void addTensorrt(int deviceNum) throws OrtException {
+            addTensorrt(OnnxRuntime.ortApiHandle,nativeHandle,deviceNum);
+        }
+
+        public void addNnapi() throws OrtException {
+            addNnapi(OnnxRuntime.ortApiHandle,nativeHandle);
+        }
+
+        public void addNuphar(boolean allowUnalignedBuffers, String settings) throws OrtException {
+            addNuphar(OnnxRuntime.ortApiHandle,nativeHandle,allowUnalignedBuffers?1:0, settings);
+        }
+
+        //ORT_API(void, OrtEnableSequentialExecution, _In_ OrtSessionOptions* options);
+        //ORT_API(void, OrtDisableSequentialExecution, _In_ OrtSessionOptions* options);
+        private native void setSequentialExecution(long apiHandle, long nativeHandle, boolean enable) throws OrtException;
```

Comment:
Ok. I've removed the comment specifying the method names. As it's a two element enum I figured that it would be ok to leave it as a boolean. If you want I can push the enum all the way through up to Java, but it's a fair amount of code. Should I rename the public method?

Reply:
We want to leave the option to allow additional execution modes, which is why we changed to an enum. One other way to do this is to have 2 separate methods setSequential and setParallel. The

Reply:
Ok, I'll do the enum tomorrow then. If the C API enum changes the Java one will need to be updated, along with the methods that do the C <-> Java in OrtJniUtil.

Reply:
Done.